### ISSUE TITLE
Restrictive transaction versioning and how to reduce resulting workload on developers.

### ISSUE THREAD

**Issue by LOLLOLOOLOL** [OmniLayer/spec#196]
> [A Master Protocol implementation] must treat addresses which broadcast messages using version numbers it does not recognize as "black holes". [...] This approach allows old versions of the Master Protocol to continue operating using the transactions they recognize without trying to parse messages of unknown meaning.

This creates a scenario in which any implementation that does not recognize version numbers may not properly consider certain addresses "black holes." It is possible for implementations to "hard fork" yet still be interoperable. This threat can be realized by the open nature of the project in implementations that lack due diligence, or are not properly overseen.

Maybe there's a way to keep a sort of "forward compatibility?" This could potentially be done by removing the requirement that revision numbers be recognized, and instead define "black hole" addresses by (whether intentionally or not) the requirement that a prior version cannot recognize a newer. In this manner, black hole addresses could be n revisions apart rather than the current requirement that they are 1 version apart. Or am I wrong to assume that an version increment is not necessarily a hard forkable event?

Looking for thoughts on this - what is the feasibility of having less strict update requirements for developers? 

tl;dr: There are unnecessarily restrictive versioning definitions. Can cause scalability issues, and serious problems with negligent implementations. Looking to open discussion on how to reduce resulting workload for developers.
